File: CodingInterviewsPythonSolutions/128. Longest Consecutive Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        mySet = set(nums)
        answer = 0
        
        for val in nums:
            if val - 1 in mySet:
                continue
            
            length = 0
            while val in mySet:
                length += 1
                val += 1
            
            answer = max(answer, length)
        
        return answer

File: CodingInterviewsPackSolutionsCPP/128. Longest Consecutive Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int sz = nums.size();
        unordered_set<int>mySet;
        for(int i = 0; i < sz; i++)
            mySet.insert(nums[i]);
        
        int answer = 0;
        for(int val:nums){
            if(mySet.count(val-1))
                continue;
            
            int len = 0;
            while(mySet.count(val)){
                len++;
                val++;
            }
            answer = max(answer,len);
        }
        return answer;
    }
};
